reposcan/database/modules_store.py

```diff
@@ -87,9 +106,119 @@ def _populate_streams(self, modules):
             cur.close()
         return modules
 
+    def _populate_rpm_artifacts(self, modules, repo_id):
+        cur = self.conn.cursor()
+        nevras_in_repo = self._get_nevras_in_repo(repo_id)
+        to_associate = set()
+        for module in modules:
+            if 'artifacts' in module:
+                for rpm in module['artifacts']:
+                    split_pkg_name = self._split_packagename(rpm)
+                    if split_pkg_name in nevras_in_repo:
+                        to_associate.add((nevras_in_repo[split_pkg_name], module['stream_id'],))
+                    else:
+                        self.logger.info('Nevra %s missing in repo %s', rpm, repo_id)
+        if to_associate:
+            execute_values(cur,
+                           """select pkg_id, stream_id from module_rpm_artifact
+                               inner join (values %s) t(pkg_id, stream_id)
+                               using (pkg_id, stream_id)
+                           """, list(to_associate), page_size=len(to_associate))
+            for row in cur.fetchall():
+                to_associate.remove((row[0], row[1],))
+        if to_associate:
+            execute_values(cur,
+                           """insert into module_rpm_artifact (pkg_id, stream_id)
+                              values %s""",
+                           list(to_associate), page_size=len(to_associate))
+        cur.close()
+        self.conn.commit()
+
+    def _populate_profiles(self, modules):
+        cur = self.conn.cursor()
+        profiles = set()
+        profile_map = {}
+        for module in modules:
+            for profile in module['profiles']:
+                profiles.add((module['stream_id'], profile,))
+        if profiles:
+            execute_values(cur,
+                           """select id, stream_id, profile_name from module_profile
+                              inner join (values %s) t(stream_id, profile_name)
+                              using (stream_id, profile_name)
+                           """, list(profiles), page_size=len(profiles))
+            for row in cur.fetchall():
```

Comment:
I think `for id, stream_id, profile_name in cur.fetchall():...` would be more clear here. You don't need to use indexing over `row` variable in next two lines.

Reply:
The same at line 164.